### PR TITLE
[LP-2504] Revoke access of ADG Admins and BU admins from /admin

### PR DESCRIPTION
**Pull request by muak50**

### CODE REVIEW

File: openedx/adg/lms/applications/admin.py [hunk 1]
```diff
@@ -89,6 +89,15 @@ class UserApplicationAdmin(admin.ModelAdmin):
...
     """
 
+    def has_change_permission(self, request, obj=None):
+        """
+        Revoke permission to edit user applications if user is not super user. As BU Admin and ADG Admin will also
+        have admin access to view user applications so we are revoking the edit permission for them.
+        """
+        if not request.user.is_superuser:
+            return False
+        return True
```
Comment (harris-irfan) [inline openedx/adg/lms/applications/admin.py:99]: we can reduce it to one line:

`return request.user.is_superuser`

**Review by zahid-arbisoft — APPROVED** [review]
Other than Harris's comment, the rest of PR looks good

**Review by harris-irfan — APPROVED** [review]
LGTM

**Review by harris-irfan — CHANGES_REQUESTED** [review]
Aren't we supposed to write unit test(s) for this method?

File: openedx/adg/lms/applications/tests/test_admin/test_user_application_admin.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    """
+    Test that only superuser is allowed to change user application.
+    """
+    assert UserApplicationAdmin.has_change_permission('self', mock_request)
+
+
+@pytest.mark.django_db
+def test_change_user_application_permission_restriction_with_only_staff_user(mock_request):
+    """
+    Test that change user application permission is not available for user who are just staff user and not superuser.
```
Comment (harris-irfan) [inline openedx/adg/lms/applications/tests/test_admin/test_user_application_admin.py:33]: Test that change user application permission is not available for users who have staff status but not superuser status.

**Review by harris-irfan — APPROVED** [review]
LGTM 👍

**Pull request merged**